### DISCUSSION TITLE
La variable "i" de un for se resetea/muere anoser que incluya un print de su puntero?

### DISCUSSION THREAD

**Discussion by Ariimaw** [IIC2133-PUC/2024-2#37] [Tarea 0] [answered]
Hola!! Es tan absurdo como suena el titulo, dejenme mostrar las sgtes screenshots:

Todo el codigo que voy a mostar esta dentro de un for con una variable "i", el mismisimo for que venia en el codigo base para manejar los eventos. 
![image](https://github.com/user-attachments/assets/3e0065d6-89ea-4b71-83f7-9002aa764402)

Sin los prints del puntero de "i":
- 
![Capture1](https://github.com/user-attachments/assets/d884b1fc-9a98-450b-8ea6-f3beb51632bd)
![Capture2](https://github.com/user-attachments/assets/3a57a07b-7a0c-4310-aa2e-c7258342ec0b)
- Se que es mucho print pero basicamente solo printeo el total de eventos aveces (7) y la mayoria de los numeros que salen son solo para traquear el "i" del for de los eventos. Donde esta destacado con amarillo se ve que el valor de "i" cambia justo despues de salir del else por alguna razon :'u pense quizas que el "m" del for dentro del else podria ser el problema entonces empece a imprimir punteros para cachar.

Con los prints del puntero de "i":
- 
![Capture3](https://github.com/user-attachments/assets/90b49c74-cc52-4c66-97bc-62ea0c18280a)
![Capture4](https://github.com/user-attachments/assets/721b0947-96f4-4e29-846d-afdbbcc3df7a)
- El unico cambio que realice en el codigo es printear la direccion de las variables "i" y "m", y por alguna razon ahora el valor de "i" no muere? osea, ahora me esta contando bien cuantos eventos han pasado y no se resetea a cero, pero no entiendo como puede posiblemente estar relacionado con printear su puntero WUUAAJJ no entiendo pq eso lo "arregla" si es el unico cambio?

Como aclaracion, uso los mismos txt en ambos casos y me asegure de compilar ambas versiones con prints y sin prints (muchas veces), no tengo otro for en el codigo que utilice una variable "i" y de verdad que me cuesta ver porque la variable del for se cambia. Tambien me asegure de que el input este bueno con los numeros correctos de eventos y puffles que me invente.

Se que probablemente se puede arreglar quizas declarando "i" antes? o eliminando ese else donde hay problemas y usar el mismo if de antes no mas (valid es para ver si se puede agregar un pinguino ono), o quizas algun arreglo este pasandose de lo que le corresponde (aunque no deberia y eso tampoco explica como un print hace que no haya problemasj ajjasdsd) pero mas me inquieta entender pq esta pasando esto, mas que si se puede arreglar o no, pq de que se puede, se que se puede, pero quiero entender que pasa </3

No muestro mas codigo pq me da wiwi que me llegue un integridad academicazaso pero ya subi a mi repo el estado actual de la cuestion si alguien se anima a ver :'u Muchas gracias de ante mano!!

**Comment by tomastrivino** [comment, marked as answer]
Holaaa!! Antes de responder con más detalle, me gustaría comentar que hay un error en el código base, en donde se define 'char color'. Véase #26. Podrías cambiarlo a 'char *color' o a 'char color[largo cualquiera]' para ver si eso soluciona algo?

**Reply by Ariimaw** [reply]
omgg comprendo! lo voy a intentar :3 Lamentablemente ta complicao pq ya tengo los structs y funciones con la forma mala JASJAJSDJAS y como no veo mucho la diferencia me voy a demorar un chiquitin en encajarlo de nuevo. Aviso en un ratito si resulta!!!! gracias!!

**Reply by waknaudt** [reply]
Que tema es ese? :0

**Reply by Ariimaw** [reply]
Bueno, efectivamente lo arreglo JSAJASJDJASJD igual me perturba que printear un puntero haya hecho una differencia pero como parece haber sido un error de como taban los strings mas que manejo de las variables mismas confio en que no es algo que pase todo el tiempo, muchas gracias!!! :333
